predicators/envs/playroom.py

```diff
@@ -924,3 +924,37 @@ def _robot_can_move(self, state: State, action: Action) -> bool:
                 for door in self._doors):
             return door == self._get_door_next_to(next_state)
         return False
+
+
+class PlayroomHardEnv(PlayroomEnv):
+    """Boring room vs playroom domain where the light on/off function is
+    arbitrary and hard."""
+    dial_on_intervals: ClassVar[List[Tuple[float, float]]] = [(0.0, 0.05),
+                                                              (0.3, 0.65),
+                                                              (0.9, 1.0)]
+
+    @classmethod
+    def get_name(cls) -> str:
+        return "playroom_hard"
+
+    def _transition_dial(self, state: State) -> State:
+        next_state = state.copy()
+        level = state.get(self._dial, "level")
+        light_is_on = False
+        for (low, high) in self.dial_on_intervals:
+            if low <= level <= high:
+                light_is_on = True
+        if not light_is_on:
```

Comment:
no action required, but to make sure I understand: we're only going to see a variety of dial values for states before the dial has been touched. after the dial is touched, it will either be set to 0.5 or 1.0 for the rest of the episode. if it was only ever 0.5 or 1.0, there's no way that we could learn the right predicate classifier, but since there's also variation in the initialization, we do have hope to learn it.

also, the reason for doing it this way is to avoid the need to continuously parameterize the dial turning option (although that would be pretty straightforward).

is that all correct?

Reply:
yeah, that is what's happening. I didn't realize this before though -- this is probably another reason why the experimental results, even using Ask All,  were not very good.